modules/core/src/hal_replacement.hpp

```diff
@@ -813,6 +813,25 @@ inline int hal_ni_minMaxIdx(const uchar* src_data, size_t src_step, int width, i
 #define cv_hal_minMaxIdx hal_ni_minMaxIdx
 //! @endcond
 
+/**
+   @brief calculates the mean and the standard deviation of array elements independently for each channel
+   @param src_data Source image
+   @param src_step Source image
+   @param width Source image dimensions
+   @param height Source image dimensions
+   @param depth Depth of source image
```

Comment:
depth should be renamed to type, because potentially the function supports multi-channel images. There should also be a parameter (for the future) to compute the whole CxC standard deviation matrix (for C-channel array) or just C-element main diagonal of the matrix. The parameter could be named 'bool diagonal_stddev' and put immediately after stddev_val.